torch/lib/THD/master_worker/common/CommandChannel.cpp

```diff
@@ -80,23 +84,88 @@ bool MasterCommandChannel::init() {
    // close listen socket
   ::close(_sockets[0]);
   _sockets[0] = -1;
+
+  std::thread error_thread(&MasterCommandChannel::errorHandler, this);
+  error_thread.detach();
```

Comment:
I think this solution is unsafe. You're detaching the thread, but it keeps reading variables that have their lifetimes bound to the CommandChannel. I think it's possible that the destructor will be called and the object will be freed, but the thread will not manage to notice the state transition. Don't detach it and figure out a way to give it a different stop signal.